### PR TITLE
Appeals API - HLR Submission Persistence

### PR DESCRIPTION
**Pull request by edmangimelli**
resolves https://vasdvp.atlassian.net/browse/DASH-153

creates a model for temporarily storing HLR Create submissions (stores headers, and json body encrypted)

### CODE REVIEW

File: modules/appeals_api/db/migrate/20200402145412_hlr_submission.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class HlrSubmission < ActiveRecord::Migration[5.2]
+  def change
+    enable_extension 'uuid-ossp'
+    enable_extension 'pgcrypto'
+
+    create_table :hlr_submissions, id: :uuid do |t|
```
Comment (charleystran) [inline modules/appeals_api/db/migrate/20200402145412_hlr_submission.rb:8]: this table should be called appeals_api_hlr_submissions or better yet appeals_api_higher_level_reviews

File: modules/appeals_api/db/migrate/20200402145412_hlr_submission.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      t.integer :status, default: 0
+      t.string :encrypted_json
```
Comment (bastosmichael) [inline modules/appeals_api/db/migrate/20200402145412_hlr_submission.rb:10]: This needs to be form_data

File: modules/appeals_api/db/migrate/20200402145412_hlr_submission.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      t.string :encrypted_json_iv
+      t.string :encrypted_headers
```
Comment (bastosmichael) [inline modules/appeals_api/db/migrate/20200402145412_hlr_submission.rb:12]: This needs to be auth_headers

File: modules/appeals_api/db/migrate/20200402145412_hlr_submission.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+      t.string :encrypted_headers_iv
+      t.string :encrypted_pdf
```
Comment (bastosmichael) [inline modules/appeals_api/db/migrate/20200402145412_hlr_submission.rb:14]: this needs to be file_data

**Comment by edmangimelli** [comment]
>This needs to be form_data
>This needs to be auth_headers
>This needs to be file_data

Just out of curiosity, are these have-to's for implementation reasons, or for sharing conventions with Claims?

File: modules/appeals_api/db/migrate/20200402145412_hlr_submission.rb (hunk 1 above)
Comment (edmangimelli) [reply modules/appeals_api/db/migrate/20200402145412_hlr_submission.rb:8]: i like it (hlr -> higher_level_review). but here's to hoping we'll never need to make a join table --i've run into problems with long table names before

**Comment by edmangimelli** [comment]
> Can we rename the table to just higher_level_reviews? That would put it in line with names of the claims_api tables

to me it seems likely that we'll want an actual HigherLevelReviews model, though. 
i'd rather not have one model for both saving/encrypting user submissions (which we want to clean up regularly) and representing HigherLevelReviews with whatever business logic we want to attach to them. I would opt for two smaller models.
thoughts?

**Comment by bastosmichael** [comment]
I think one persisted model per appeals type is perfectly fine, also @charleystran I think the reason I wanted to persist the PDF itself was for debugging purposes, what happens if an issue happens with a bad pdf and we can't reproduce it because we don't have the original pdf that was generated to see for ourselves? Your call on this since you're leading this charge but something to consider on the issue of PDF persistence.

**Comment by charleystran** [comment]
@bastosmichael I think persisting the PDF should be its own ticket if we want to do it because we need to also create jobs to cleanup old pdf's.

**Pull request merged**